### ISSUE TITLE
Fisheye 200 degree video play as regular 180

### ISSUE THREAD

**Issue by hjs0007** [xbapps/xbvr#338]
Player usually use naming convention to determine how to play a file. Because the filename returned to the player in the json from deo portal is not the original name, the player can't detect how to play a video and seem like to fallback to VR180 automatically. You should set stereoMode and screenType parameters (see https://deovr.com/doc#single-video-deeplink) based on original filename.

**Comment by theRealKLH** [comment]
if the video could be downloaded in multiple versions, how do you choose the "original filename"? outside of video preview generation is xbvr looking inside the video to determine the type of encode/format?

**Comment by hjs0007** [comment]
By original filename, I mean the filename I use on my local storage.

**Comment by theRealKLH** [comment]
Gotcha. But my original filename doesn't specify VR180 or VR200 or even SBS, TB, OU, etc (playing devil's advocate).

**Comment by hjs0007** [comment]
You should. This is how most players detect how to play the file.

To return to the main issue, I tried to fix the issue. I see that SLR is sending this json:

```json
 "stereoMode": "sbs2l",
 "projection": "fisheye_regular",
```
Unfortunately, when I send the same data, Deo is still playing the video as 180 SBS. Seem like Deo is using an SLR  "proprietary" extension again...

**Comment by theRealKLH** [comment]
but xbvr isn't a player.

**Comment by hjs0007** [comment]
` are you saying xbvr is sending something other than the true filename`

Yep. XBVR return an url with the original name that is gone. Also, even if the name would be retained, I'm not sure if the detection based on name would override the json data. This is why this is important that xbvr return the right configuration (based on original name or technical data about the video file).  But I discovered that it seem like the deo api doesn't allow it right now. It works with SLR because Deo use a proprietary api for them.

**Comment by github-userx** [comment]
Noob here, not sure if this might help but see from deovr dev:

>Content producers can enjoy ZCam K1 Pro fish-eye mesh support. That way you can get original 220º FOV with no distortion caused by equirectangular conversion. **Simply add "MKX220" into filename and it will be recognised automatically.**
>
> Source: https://i.reddit.com/r/xvr/comments/gl48g6/deovr_update_headtracking_axes_default_180_sbs/

**Comment by flightsim73** [comment]
Any other thoughts on this? Is there a way to parse the filename and then update the screenType?
https://deovr.com/doc

"stereoMode" can be set to "sbs" for side by side stereoscopic layout, "tb" for top-bottom layout or "off" for monoscopic videos.
`"stereoMode": "tb"`

"screenType" can be set to "flat" for a 2D quad display inside of VR, "dome" for 180° videos, "fisheye" for fisheye lens videos and "sphere" for 360° videos, “220” for 220° mesh and “250” for 250° mesh.
`"screenType": "sphere"`

**Comment by hjs0007** [comment]
Fixed in #433

**Comment by meihaco** [comment]
You can also convert SLR fisheye (200° or 190°) to normal 180° video with FFmpeg:

```bash
ffmpeg -i input.mp4 -filter:v "v360=input=fisheye:ih_fov=200:iv_fov=200:output=hequirect:in_stereo=sbs:out_stereo=sbs" -map 0 -c:a copy -c:s copy -c:v libx264 -preset slow -pix_fmt yuv420p output.mp4
```

- Replace `input.mp4` with your file name.
- Change `ih_fov=200` and `iv_fov=200` to `190` if your original is 190°.
- `output.mp4` will be a standard 180° equirectangular video, playable anywhere.

Done! After conversion, you don't need the SLR app.

I feel like this would be useful for many people here, since this topic pop-up first when you trying to google this issue.
It's the easiest and simplest solution.
